diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9a24c4..8d2499e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,9 +2,9 @@ cmake_minimum_required(VERSION 3.28)
 
 # project / version
 set(REDUMPER_PROJECT_NAME "redumper" CACHE STRING "Project name")
-string(TIMESTAMP REDUMPER_VERSION_MAJOR "%Y")
-string(TIMESTAMP REDUMPER_VERSION_MINOR "%m")
-string(TIMESTAMP REDUMPER_VERSION_PATCH "%d")
+set(REDUMPER_VERSION_MAJOR "%Y" CACHE STRING "Commit Year")
+set(REDUMPER_VERSION_MINOR "%m" CACHE STRING "Commit Month")
+set(REDUMPER_VERSION_PATCH "%d" CACHE STRING "Commit Day")
 set(REDUMPER_VERSION_BUILD "LOCAL" CACHE STRING "Version patch")
 project(${REDUMPER_PROJECT_NAME} VERSION "${REDUMPER_VERSION_MAJOR}.${REDUMPER_VERSION_MINOR}.${REDUMPER_VERSION_PATCH}")
 
